components/admin_console/team_channel_settings/channel/details/channel_moderation.tsx

```diff
@@ -0,0 +1,331 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import {isNil} from 'lodash';
+import classNames from 'classnames';
+import {ChannelModeration as ChannelPermissions} from 'mattermost-redux/types/channels';
+
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+
+import {t} from 'utils/i18n';
+
+import AdminPanel from 'components/widgets/admin_console/admin_panel';
+import CheckboxCheckedIcon from 'components/widgets/icons/checkbox_checked_icon.jsx';
+t('admin.channel_settings.channel_moderation.title');
+t('admin.channel_settings.channel_moderation.subtitle');
+t('admin.channel_settings.channel_moderation.permissions');
+t('admin.channel_settings.channel_moderation.guests');
+t('admin.channel_settings.channel_moderation.members');
+const formattedMessages: any = {
+    create_post: {
+        title: {
+            id: t('admin.channel_settings.channel_moderation.createPosts'),
+            defaultMessage: 'Create Posts'
+        },
+        description: {
+            id: t('admin.channel_settings.channel_moderation.createPostsDesc'),
+            defaultMessage: 'The ability for members and guests to create posts in the channel.'
+        },
+        disabled_guests: {
+            id: t('admin.channel_settings.channel_moderation.createPosts.disabledGuest'),
+            defaultMessage: 'Create posts for guests are disabled in [{scheme_name}](../permissions/{scheme_link}).'
+        },
+        disabled_members: {
+            id: t('admin.channel_settings.channel_moderation.createPosts.disabledMember'),
+            defaultMessage: 'Create posts for members are disabled in [{scheme_name}](../permissions/{scheme_link}).'
+        },
+        disabled_both: {
+            id: t('admin.channel_settings.channel_moderation.createPosts.disabledBoth'),
+            defaultMessage: 'Create posts for members and guests are disabled in [{scheme_name}](../permissions/{scheme_link}).'
+        }
+    },
```

Comment:
I think every key that is not mapped to something directly returning from the API should be in `camelCase` thoughts? 
or how bought nesting the keys like 
```
disabled: {
    guests: {
        ...
    },
    members: {
        ...
    },
}
```

Reply:
Good catch on the camel case. I made the keys not mapped to api camel case. I didn't want to nest it because it would complicate things a little bit more. Thanks for that.